react-components/src/hooks/useSearchAssetsMapped360Annotations.tsx

```diff
@@ -0,0 +1,149 @@
+
+          return isInName || isInDescription;
+        }) ?? [];
+
+      return filteredSearchedAssets;
+    },
+    {
+      staleTime: Infinity,
+      enabled: isFetched && assetMappings !== undefined
+    }
+  );
+};
+
+async function getAssetsMapped360Annotations(
+  sdk: CogniteClient,
+  siteIds: string[]
+): Promise<Asset[]> {
+  const fileIdsList = await get360ImagesFileIds(siteIds, sdk);
+  const image360Annotations = await get360ImageAnnotations(fileIdsList, sdk);
+  const result = await get360AnnotationAssets(image360Annotations, sdk);
+
+  return result;
+}
+
+async function get360AnnotationAssets(
+  image360Annotations: AnnotationModel[],
+  sdk: CogniteClient
+): Promise<Asset[]> {
+  const annotationMapping = image360Annotations.map((annotation) => ({
+    assetId:
+      (annotation.data as AnnotationsBoundingVolume).assetRef?.id ??
+      (annotation.data as AnnotationsBoundingVolume).assetRef?.externalId ??
+      undefined
+  }));
+
+  const uniqueAnnotationMapping = uniqBy(annotationMapping, 'assetId');
+
+  const assets = chunk(uniqueAnnotationMapping, 1000).map(async (uniqueAssetsChunk) => {
+    const retrievedAssets = await sdk.assets.retrieve(
+      uniqueAssetsChunk.map(({ assetId }) => {
+        if (typeof assetId === 'number') {
+          return { id: assetId };
+        } else if (typeof assetId === 'string') {
+          return { externalId: assetId };
+        } else {
+          throw new Error(`Invalid assetId: ${assetId}`);
+        }
+      }),
+      { ignoreUnknownIds: true }
+    );
+    return retrievedAssets;
+  });
+
+  return (await Promise.all(assets)).flat();
+}
+
+async function get360ImagesFileIds(siteIds: string[], sdk: CogniteClient): Promise<number[]> {
+  const fileIdListPromises = siteIds.map(async (siteId) => {
+    const req: FileFilterProps = {
+      metadata: { site_id: siteId }
+    };
+    const fileIds = await listFileIds(req, sdk);
+    return fileIds;
+  });
+
+  const fileIdsList = (await Promise.all(fileIdListPromises)).flat();
+
+  return fileIdsList;
+}
+
+async function get360ImageAnnotations(
+  fileIdsList: number[],
+  sdk: CogniteClient
+): Promise<AnnotationModel[]> {
+  const annotationArray = await Promise.all(
+    chunk(fileIdsList, 1000).map(async (fileIdsChunk) => {
+      const filter: AnnotationFilterProps = {
+        annotatedResourceIds: fileIdsChunk.map((id) => ({ id })),
+        annotatedResourceType: 'file',
+        annotationType: 'images.AssetLink'
+      };
+      const { items: annotations } = await sdk.annotations.list({
+        filter
+      });
+      return annotations;
+    })
+  );
+
+  return annotationArray.flatMap((annotations) => annotations);
+}
+
+async function listFileIds(filter: FileFilterProps, sdk: CogniteClient): Promise<number[]> {
+  const req = { filter, limit: 1000 };
+  const map = await sdk.files.list(req).autoPagingToArray({ limit: Infinity });
+
+  const fileInfo = await Promise.all(map.flat());
```

Comment:
I don't think you need to use Promise.all here, see how we use the same method for point cloud annotation works without that part and flattening. 

https://github.com/cognitedata/reveal/blob/master/viewer/packages/data-providers/src/pointcloud-stylable-object-providers/CdfPointCloudStylableObjectProvider.ts (fetchAnnotations method)